src/main/java/io/cdap/delta/bigquery/BigQueryTarget.java

```diff
@@ -93,13 +105,33 @@ public void initialize(DeltaTargetContext context) throws Exception {
       .build()
       .getService();
 
-    long maximumExistingSequenceNumber =
-      BigQueryUtils.getMaximumExistingSequenceNumber(context.getAllTables(), project, conf.getDatasetName(),
-                                                     bigQuery, encryptionConfig, MAX_TABLES_PER_QUERY);
+    RetryPolicy<Object> retryPolicy = createBaseRetryPolicy()
+            .handleIf(throwable -> {
+              if (throwable instanceof BigQueryException) {
+                BigQueryException t = (BigQueryException) throwable;
+                int code = t.getCode();
+                String reason = t.getError() != null ? t.getError().getReason() : null;
+                boolean isRateLimitExceeded =  RATE_LIMIT_EXCEEDED_CODES.contains(code)
+                        && RATE_LIMIT_EXCEEDED_REASON.equals(reason);
+                boolean isBillingTierLimitExceeded = code == BILLING_TIER_LIMIT_EXCEEDED_CODE
+                        && BILLING_TIER_LIMIT_EXCEEDED_REASON.equals(reason);
+                return (t.isRetryable() || isRateLimitExceeded || isBillingTierLimitExceeded);
```

Comment:
nit: remove parenthesis
```suggestion
                return t.isRetryable() || isRateLimitExceeded || isBillingTierLimitExceeded;
```